File: app/src/main/java/com/stedi/randomimagegenerator/app/view/fragments/base/BaseFragment.java
package com.stedi.randomimagegenerator.app.view.fragments.base;

import android.support.annotation.NonNull;

import com.stedi.randomimagegenerator.app.App;
import com.stedi.randomimagegenerator.app.view.activity.base.BaseActivity;

public abstract class BaseFragment extends LifeCycleFragment {
    public BaseActivity getBaseActivity() {
        return (BaseActivity) getActivity();
    }

    public boolean checkForPermission(@NonNull String permission, int requestCode) {
        return getBaseActivity().checkForPermission(permission, requestCode);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        ((App) getActivity().getApplicationContext()).getLeakWatcher().watch(this);
    }
}

File: app/src/main/java/com/stedi/randomimagegenerator/app/view/fragments/base/BaseFragment.kt
package com.stedi.randomimagegenerator.app.view.fragments.base

import com.stedi.randomimagegenerator.app.other.getApp
import com.stedi.randomimagegenerator.app.view.activity.base.BaseActivity
import com.stedi.randomimagegenerator.app.view.components.RequireViewModel

abstract class BaseFragment : LifeCycleFragment(), RequireViewModel {

    fun checkForPermission(permission: String, requestCode: Int): Boolean {
        val activity = activity ?: return false
        return (activity as BaseActivity).checkForPermission(permission, requestCode)
    }

    override fun onDestroy() {
        super.onDestroy()
        activity?.apply { getApp().leakWatcher.watch(this) }
    }
}